src/coordinator/block/types.go

```diff
@@ -43,36 +43,101 @@ type SeriesMeta struct {
 	Name string
 }
 
-// Bounds are the time bounds
+// Bounds are the time bounds, start time is inclusive but end is exclusive
 type Bounds struct {
 	Start    time.Time
-	End      time.Time
+	Duration time.Duration
 	StepSize time.Duration
 }
 
 // TimeForIndex returns the start time for a given index assuming a uniform step size
 func (b Bounds) TimeForIndex(idx int) (time.Time, error) {
```

Comment:
Maybe this should return `(time.Time, bool)` so instead of relying on the error if it's out of bounds, we rely on a true/false